Framework/DataHandling/src/LoadMuonNexusV2.cpp

```diff
@@ -145,20 +150,21 @@ void LoadMuonNexusV2::execLoader() {
   NXRoot root(m_filename);
   NXEntry entry = root.openEntry(NeXusEntry::RAWDATA);
   isEntryMultiPeriod(entry);
-  if (m_isFileMultiPeriod) {
-    throw std::invalid_argument(
-        "Multiperiod nexus files not yet supported by LoadMuonNexusV2");
-  }
-
   // Execute child algorithm LoadISISNexus2
   auto outWS = runLoadISISNexus();
 
   // Check if single or multi period file and create appropriate loading
   // strategy
   if (m_multiPeriodsLoaded) {
-    // Currently not implemented
-    throw std::invalid_argument(
-        "Multiperiod nexus files not yet supported by LoadMuonNexusV2");
+    WorkspaceGroup_sptr workspaceGroup =
+        std::dynamic_pointer_cast<WorkspaceGroup>(outWS);
+    m_loadMuonStrategy = std::make_unique<MultiPeriodLoadMuonStrategy>(
+        g_log, m_filename, entry, workspaceGroup);
+    auto numberHistograms =
+        std::dynamic_pointer_cast<Workspace2D>(workspaceGroup->getItem(0))
```

Comment:
Could we store a reference to this pointer:
`std::shared_ptr<Workspace2D> firstWsInGroup`

Since we also do the same cast again below in the single case